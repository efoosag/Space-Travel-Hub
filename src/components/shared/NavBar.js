import React from 'react';
import { Link } from 'react-router-dom';

const NavBar = () => (
  <header className="header">
    <nav className="navbar">
      <div>
        <h2>Space Travelers Hub </h2>
      </div>
      <ul className="nav-menu list">
        <li className="nav-link">
          <Link to="/">Rockets</Link>
        </li>
        <li className="nav-link">
          <Link to="/missions"> Missions</Link>
        </li>
        <li className="nav-link border-left">
          <Link to="/profile">My Profile</Link>
        </li>
      </ul>
    </nav>
  </header>
);

export default NavBar;
