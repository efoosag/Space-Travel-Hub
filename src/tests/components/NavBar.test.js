import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import NavBar from '../../components/shared/NavBar';

describe('Testing NavBar component', () => {
  it('Testing with snapshot', () => {
    const tree = renderer.create(
      <Router><NavBar /></Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
