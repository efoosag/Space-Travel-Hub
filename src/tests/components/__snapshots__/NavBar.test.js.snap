// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NavBar component Testing with snapshot 1`] = `
<header
  className="header"
>
  <nav
    className="navbar"
  >
    <div
      className="logo"
    >
      <img
        alt="logo"
        src="space-logo.png"
      />
      <h2>
        Space Travelers Hub 
      </h2>
    </div>
    <ul
      className="nav-menu list"
    >
      <li
        className="nav-link"
      >
        <a
          aria-current="page"
          className="nav-link-active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="nav-link"
      >
        <a
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="nav-link border-left"
      >
        <a
          href="/profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
